peer/brontide.go

```diff
@@ -649,7 +649,35 @@ func (p *Brontide) Start() error {
 	go p.writeHandler()
 	go p.readHandler()
 	go p.channelManager()
-	go p.pingHandler()
+
+	var (
+		lastBlockHeader           *wire.BlockHeader
+		lastSerializedBlockHeader [wire.MaxBlockHeaderPayload]byte
+	)
+	newPingPayload := func() []byte {
```

Comment:
suggestion for minimising indentation & for opting to indent error paths rather than success paths:

```
newPingPayload := func() []byte {
		// We query the BestBlockHeader from our BestBlockView each time
		// this is called, and update our serialized block header if
		// they differ.  Over time, we'll use this to disseminate the
		// latest block header between all our peers, which can later be
		// used to cross-check our own view of the network to mitigate
		// various types of eclipse attacks.
		header, err := p.cfg.BestBlockView.BestBlockHeader()
		if err != nil || header == lastBlockHeader {
			return lastSerializedBlockHeader[:]
		}

		buf := bytes.NewBuffer(lastSerializedBlockHeader[0:0])
		if err = header.Serialize(buf); err != nil {
			p.log.Warn("unable to serialize current block" +
				"header for ping payload generation." +
				"This should be impossible and means" +
				"there is an implementation bug.")
		} else {
			lastBlockHeader = header
		}

		return lastSerializedBlockHeader[:]
	}
```